lib/puppet/provider/consul_acl/default.rb

```diff
@@ -23,14 +16,14 @@ def self.list_resources
     uri = URI('http://localhost:8500/v1/acl')
     http = Net::HTTP.new(uri.host, uri.port)
 
-    path=uri.request_uri + '/list'
+    path=uri.request_uri + "/list?token=#{acl_api_token}"
     req = Net::HTTP::Get.new(path)
     res = http.request(req)
 
     if res.code == '200'
       acls = JSON.parse(res.body)
     else
-      acls = []
+      raise(Puppet::Error,"Cannot retrieve ACLs: invalid return code #{res.code} uri: #{path} body: #{req.body}")
```

Comment:
It shouldn't be an issue, since it's no longer part of a prefetch in this version.